SVBoxSlider texture generation performance improvement using compute shader

grenqa opened the pull request:
13 ms => 1 ms. The frame time is so long because the deep profile option is enabled. In fact, they are much smaller. In any case, compute shader is faster than cpu in 13 times.

judah4 commented:
Good changes but it looks like the prefab in the testing scene broke. Fix it and I'll merge it in.

grenqa commented:
I managed to make it work on unity 2017.1, 2018.2 and 2018.3, however when I drag the prefab, a strange error occurs: Assertion failed: Assertion failed on expression: 'modifications.empty ()'. It appears only in the editor, and it seems that it does not break anything. I do not know what to do with it. I changed only 1 file and added a folder with a shader. By the way, it turns out I was looking at the wrong function call in the profiler and the shader does not generate the texture in 1 ms, but in 0.02ms!

judah4 commented:
Nice! I like the speed! Looks good, I'm merging it in. Not sure about the assert, I didn't get it on my computer with a fresh project download.
